test/periphery/CardPaymentProcessorUpgradeable.test.ts

```diff
@@ -0,0 +1,1421 @@
+        payment.amount,
+        authorizationId,
+        correlationId
+      )).to.changeTokenBalances(
+        brlcMock,
+        [cardPaymentProcessor, payment.account],
+        [+payment.amount, -payment.amount]
+      ).and.to.emit(
+        cardPaymentProcessor,
+        "MakePayment"
+      ).withArgs(
+        authorizationId,
+        correlationId,
+        payment.account.address,
+        payment.amount,
+        payment.revocationCounter || 0
+      );
+      payment.status = PaymentStatus.Uncleared;
+      await checkCardPaymentProcessorState([payment]);
+    });
+
+    it("Is reverted if the payment authorization ID already exists", async () => {
+      await makePayments([payment]);
+      const otherMakingPaymentCorrelationsId: string = createBytesString(
+        payment.makingPaymentCorrelationId + 1,
+        BYTES16_LENGTH
+      );
+      await expect(cardPaymentProcessor.connect(payment.account).makePayment(
+        payment.amount + 1,
+        authorizationId,
+        otherMakingPaymentCorrelationsId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_ALREADY_EXISTS);
+    });
+  });
+
+  describe("Function 'clearPayment()'", async () => {
+    let payment: TestPayment;
+    let admin: SignerWithAddress;
+    let authorizationId: string;
+
+    beforeEach(async () => {
+      payment = {
+        authorizationId: 123,
+        account: user1,
+        amount: 234,
+        status: PaymentStatus.Nonexistent,
+        makingPaymentCorrelationId: 345,
+      };
+      admin = user2;
+      authorizationId = createBytesString(payment.authorizationId, BYTES16_LENGTH);
+      await setUpContractsForPayments([payment]);
+      await addAccountToCardPaymentProcessorWhitelist(admin);
+      await makePayments([payment]);
+    });
+
+    it("Is reverted if the contract is paused", async () => {
+      await proveTx(cardPaymentProcessor.setPauser(deployer.address));
+      await proveTx(cardPaymentProcessor.pause());
+      await expect(cardPaymentProcessor.connect(admin).clearPayment(
+        authorizationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_CONTRACT_IS_PAUSED);
+    });
+
+    it("Is reverted if the caller is not whitelisted", async () => {
+      await expect(cardPaymentProcessor.connect(deployer).clearPayment(
+        authorizationId
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_ACCOUNT_IS_NOT_WHITELISTED);
+    });
+
+    it("Is reverted if the payment authorization ID is zero", async () => {
+      await expect(cardPaymentProcessor.connect(admin).clearPayment(
+        ZERO_BYTES16_STRING
+      )).to.be.revertedWith(REVERT_MESSAGE_IF_PAYMENT_AUTHORIZATION_ID_IS_ZERO);
+    });
+
+    it("Is reverted if the payment with the provided authorization ID does not exist", async () => {
+      await expect(cardPaymentProcessor.connect(admin).clearPayment(
+        createBytesString(payment.authorizationId + 1, BYTES16_LENGTH)
```

Comment:
Codacy has a fix for the issue: [Insert `····`](https://app.codacy.com/gh/cloudwalk/brlc-token/pullRequest?prid=9781596)
```suggestion
            createBytesString(payment.authorizationId + 1, BYTES16_LENGTH)
```